lib/block/operation.go

```diff
@@ -313,26 +309,15 @@ func GetBlockOperation(st *storage.LevelDBBackend, hash string) (bo BlockOperati
 	return
 }
 
-func GetBlockOperationWithIndex(st *storage.LevelDBBackend, hash string, opIndex int) (bo BlockOperation, err error) {
-	var found = false
-	iterFunc, closeFunc := GetBlockOperationsByTx(st, hash, nil)
-	for idx := 0; idx <= opIndex; idx++ {
-
-		o, hasNext, _ := iterFunc()
-		if !hasNext {
-			break
-		}
-		if idx == opIndex {
-			found = true
-			bo = o
-			break
-		}
-	}
-	if !found {
-		err = errors.OperationNotFound
+// Looks up the operation referenced by `txHash`, then get the operation's hash from it
+func GetBlockOperationByIndex(st *storage.LevelDBBackend, txHash string, opIndex int) (BlockOperation, error) {
+	if bt, err := GetBlockTransaction(st, txHash); err != nil {
+		return BlockOperation{}, err
+	} else if opIndex >= len(bt.Operations) {
```

Comment:
Unlike the existing code, `opIndex` can be under 0, so we need to check.
```go
} else if opIndex < 0 || opIndex >= len(bt.Operations) {
```

Reply:
In which case could it be `< 0` ?

Reply:
@Geod24 `GetBlockOperationByIndex()` is called at `GetOperationsByTxHashOpIndexHandler` and it's `opindex` is just treated as `int`. Client can set the `opindex` as negative.

Reply:
Good point. A shame the type system doesn't allow us to express this :(